### DISCUSSION TITLE
How to dynamically change the expo-platform in the requestHeaders?

### DISCUSSION THREAD

**Discussion by rafcremer** [expo/expo#24728] [Archived: Questions] [answered]
Is there a way to dynamically change the `expo-platform` in the updates.requestHeaders of the app.jon/app.config.ts ?
I am only allowed to set is as either `ios` or `android` which makes it work only for ios or android but not both.

```
updates: {
        url: '<utl>',
        requestHeaders: {
            'expo-runtime-version': '1.0.0',
            'expo-channel-name': 'develop',
            'expo-platform': 'ios'
        }
    }

**Comment by wschurman** [comment]
Hi @rafcremer, what's your use case for this functionality (setting these headers in the app config)?

The `expo-runtime-version` and `expo-platform` headers are automatically added by the expo-updates library.

**Reply by rafcremer** [reply]
Hi @wschurman, I am using the `checkForUpdateAsync` method from the expo-updates library which requires settings the headers manually, otherwise I get an error.

The reason why I need this is because I implemented a button called "Install Updates" only visible in test builds for internal test users and only enabled if any updates are available.
If any new update is available then by tapping on the button the app downloads them with `fetchUpdateAsync` and reload the app with `reloadAsync`.
It makes it easier for our internal test users to see if an update for the app is available or not and to download it, and it works really well!! but only for either iOS or Android right now, not both.

**Reply by wschurman** [reply, marked as answer]
`checkForUpdateAsync` should automatically be adding the `expo-runtime-version` and `expo-platform` headers on both platforms.

[iOS](https://github.com/expo/expo/blob/main/packages/expo-updates/ios/EXUpdates/AppLoader/FileDownloader.swift#L295) and [Android](https://github.com/expo/expo/blob/main/packages/expo-updates/android/src/main/java/expo/modules/updates/loader/FileDownloader.kt#L767) code references (these are called from `checkForUpdateAsync`).

**Reply by rafcremer** [reply]
Thanks @wschurman that worked! For some reason I had issues before when just leaving expo-platform out of it or for not having requestHeaders at all.
